types/brittle.d.ts

```diff
@@ -9,15 +9,28 @@ declare module 'brittle' {
   type AnyErrorConstructor = new () => Error
 
   interface ExceptionAssertion {
+    (
+      fn: Promise<unknown> | (() => Promise<unknown>),
```

Comment:
technically `fn` is kind of a misleading parameter name in the case of providing a promise, but didn't think that creating another overload made sense vs using a union